packages/gatsby-plugin-utils/src/joi.ts

```diff
@@ -1,8 +1,18 @@
 import joi from "joi"
-import { PluginOptionsSchemaJoi } from "./utils/plugin-options-schema-joi-type"
+import { Root } from "joi"
 
-export * from "./utils/plugin-options-schema-joi-type"
-export const Joi: PluginOptionsSchemaJoi = joi.extend({
+export interface GatsbyPluginOptionsJoi extends Root {
```

Comment:
@mxstbr This *should* accomplish the same thing you were doing before without copying the type files. But since the `dotenv` stuff isn't being implemented I commented it out.

Reply:
I don't think this is the same. Here's what people are able to do with our extensions:

```JS
field: Joi.string().dotenv("SOME_NAME")
```

I think what your types are specifying is different, they specify that they can do this, which is incorrect:

```js
field: Joi.dotenv("SOME_NAME"),
```

Am I missing something?

Reply:
Ah, cause I'm not modifying the `AnySchema`? That would make sense...I'll do more research, there's got to be a better way than re typing the whole thing.

Reply:
I thought so too but unfortunately what we need is `Joi.Root` to accept a generic, which it does not. Something like this:

```JS
type OurCustomAnySchema extends AnySchema {
  dotenv...
}

const CustomJoi: Joi.Root<OurCustomAnySchema> = Joi.extend...
```